dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Link> convertDotConnectMapToLink(final List<Map<String,String>> results)
+            throws ParseException {
+
+        List<Link> ret;
+
+        ret = new ArrayList<>();
+
+        if(results != null && !results.isEmpty()){
+            for (Map<String, String> map : results) {
+                final Link link = new Link();
+                link.setInode(map.get("inode"));
+
+                if (map.get("show_on_menu") != null && !map.get("show_on_menu").isEmpty()){
+                    link.setShowOnMenu(Boolean.parseBoolean(map.get("show_on_menu")));
+                }
+
+                link.setTitle(map.get("title"));
+
+                if (map.get("mod_date") != null && !map.get("mod_date").isEmpty()){
+                    link.setModDate(df.parse(map.get("mod_date")));
+                }
+
+                link.setModUser(map.get("mod_user"));
+
+                if (map.get("sort_order") != null && !map.get("sort_order").isEmpty()){
+                    link.setSortOrder(Integer.parseInt(map.get("sort_order")));
+                }
+
+                link.setFriendlyName(map.get("friendly_name"));
+
+                link.setIdentifier(map.get("identifier"));
+
+                link.setProtocal(map.get("protocal"));
+
+                link.setUrl(map.get("url"));
+
+                link.setTarget(map.get("target"));
+
+                link.setInternalLinkIdentifier(map.get("internal_link_identifier"));
+
+                link.setLinkType(map.get("link_type"));
+
+                link.setLinkCode(map.get("link_code"));
+
+                ret.add(link);
+            }
+        }
+
+
+        return ret;
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Identifier> convertDotConnectMapToIdentifier(final List<Map<String,String>> results)
+            throws ParseException {
+        final List<Identifier> ret = new ArrayList<>();
+        if(results == null || results.size()==0){
+            return ret;
+        }
+
+        for (Map<String, String> map : results) {
+            final Identifier i = new Identifier();
+            i.setAssetName(map.get("asset_name"));
+            i.setAssetType(map.get("asset_type"));
+            i.setHostId(map.get("host_inode"));
+            i.setId(map.get("id"));
+            i.setParentPath(map.get("parent_path"));
+
+            if (map.get("syspublish_date") != null && !map.get("syspublish_date").isEmpty()){
+                i.setSysPublishDate(df.parse(map.get("syspublish_date") ));
+            }
+
+            if (map.get("sysexpire_date") != null && !map.get("sysexpire_date").isEmpty()){
+                i.setSysExpireDate(df.parse(map.get("sysexpire_date") ));
+            }
+
+            ret.add(i);
+        }
+        return ret;
+    }
+
+    public static List<Folder> convertDotConnectMapToFolder(final List<Map<String,String>> results)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)